derive_builder_core/src/build_method.rs

```diff
@@ -65,6 +68,11 @@ impl<'a> ToTokens for BuildMethod<'a> {
             BuilderPattern::Immutable => quote!(&self),
         };
         let doc_comment = &self.doc_comment;
+        let default = if let Some(ref expr) = self.struct_default { 
+            quote!(let __default : #target_ty = #expr;)
```

Comment:
Possible optimization: Check if all fields are explicitly initialized and avoid allocating the extra object if it won't be used.